app/src/routes/v2/template.js

```diff
@@ -0,0 +1,76 @@
+const Problem = require('api-problem');
+const templateRouter = require('express').Router();
+
+const { findAndRender, getFromCache } = require('../../components/carboneCopyApi');
+const FileCache = require('../../components/fileCache');
+const { upload } = require('../../components/upload');
+const { truthy } = require('../../components/utils');
+const { middleware } = require('../../components/validation');
+
+const fileCache = new FileCache();
+
+/** Returns the rendered report from cache */
+templateRouter.post('/', upload, async (req, res) => {
+  console.log('Template upload');
+
+  if (!req.file) {
+    return new Problem(422, { detail: 'Template file is missing or malformed.' }).send(res);
+  }
+
+  // TODO: If `carbone.uploadCount` is greater than 1, check `req.files` array
+  const result = await fileCache.move(req.file.path, req.file.originalname);
+  if (!result.success) {
+    return new Problem(result.errorType, { detail: result.errorMsg }).send(res);
+  } else {
+    res.setHeader('X-Template-Hash', result.hash);
+    return res.send(result.hash);
+  }
+});
+
+templateRouter.post('/render', middleware.validateTemplate, async (req, res) => {
+  console.log('Template upload and render');
```

Comment:
Should we remove this console.log?

Reply:
This is an artifact from npm package shuffling. If/when we get to json log refactor, I'm expecting this to be emitted as a log.verbose notification.